nbsite/pyodide/__init__.py

```diff
@@ -197,115 +246,158 @@ class PyodideDirective(Directive):
         'skip-embed': _option_boolean
     }
 
-    _current_source = None
-    _current_context = {}
-    _current_count = 0
-    _current_process = None
-    _conn = None
+    _exec_state = defaultdict(lambda: {
+        'source': None,
+        'context': {},
+        'count': 0,
+        'total': None,
+        'process': None,
+        'conn': None,
+        'cache': {},
+        'hash': None
+    })
 
     @classmethod
     def _execution_process(cls, pipe):
         """
         Process execution loop to run in separate process that is used
         to evaluate code.
         """
-        while True:
-            msg = pipe.recv()
-            if msg['type'] == 'close':
-                break
-            elif msg['type'] != 'execute':
-                continue
-            stdout = io.StringIO()
-            stderr = io.StringIO()
-            code = msg['code']
-            with set_resource_mode('cdn'):
-                try:
-                    out = exec_with_return(code, stdout=stdout, stderr=stderr)
-                except Exception:
-                    out = None
-                if (isinstance(out, (Model, Viewable, Viewer)) or
-                    any(pane.applies(out) for pane in CONVERT_PANE_TYPES)):
-                    _, content = _model_json(as_panel(out), msg['target'])
-                    mime_type = 'application/bokeh'
-                elif out is not None:
+        async def loop():
+            while True:
+                msg = pipe.recv()
+                if msg['type'] == 'close':
+                    break
+                elif msg['type'] != 'execute':
+                    continue
```

Comment:
```suggestion
                    await asyncio.sleep(0.01)
                    continue
```

Just to give it a bit of breathing room.

Reply:
Not needed I think, `recv` is a blocking API. So there's no worry of thrashing through the loop here.